internal/core/metadata/application/deviceprofile.go

```diff
@@ -206,6 +207,126 @@ func PatchDeviceProfileBasicInfo(ctx context.Context, dto dtos.UpdateDeviceProfi
 	return nil
 }
 
+func DeleteDeviceResourceByName(profileName string, resourceName string, dic *di.Container) errors.EdgeX {
+	if profileName == "" {
+		return errors.NewCommonEdgeX(errors.KindContractInvalid, "profile name is empty", nil)
+	}
+	if resourceName == "" {
+		return errors.NewCommonEdgeX(errors.KindContractInvalid, "resource name is empty", nil)
+	}
+
+	strictProfileChanges := container.ConfigurationFrom(dic.Get).Writable.ProfileChange.StrictDeviceProfileChanges
+	if strictProfileChanges {
+		return errors.NewCommonEdgeX(errors.KindServiceLocked, "profile change is not allowed", nil)
```

Comment:
`profile change is not allowed when StrictDeviceProfileChanges config is enabled`